src/tls/keypair.h

```diff
@@ -164,54 +240,23 @@ namespace tls
             throw std::logic_error("Could not generate ECDSA keypair");
       }
 
-#if CURVE_CHOICE_SECP256K1_BITCOIN
-      if (
-        mbedtls_mpi_write_binary(&(mbedtls_pk_ec(*key)->d), c4_priv, PK_SIZE) !=
-        0)
-        throw std::logic_error("Could not extract raw private key");
-#endif
-    }
-
-    KeyPair(const KeyPair&) = delete;
-    KeyPair(KeyPair&& other)
-    {
-      key = std::move(other.key);
-      other.key = nullptr;
-#if CURVE_CHOICE_SECP256K1_BITCOIN
-      ctx = std::move(other.ctx);
-      other.ctx = nullptr;
-#endif
+      if (get_ec_from_context(*key) != ec)
+      {
+        throw std::logic_error("Created key and received unexpected type");
```

Comment:
Might be good to print the mismatching value as well.

Reply:
I've now added the return codes/unexpected values to all logic errors thrown from `keypair.h`.